libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/OAuthClient.java

```
@@ -84,54 +89,85 @@ public URI MakeUri(String uri, HashMap<String, String> queryStrings) throws URIS
+            HashMap <String, String> qstrings = new HashMap<>();
+            qstrings.put("userId", userId);
+            qstrings.put("connectionName", connectionName);
+            if (!StringUtils.isBlank(magicCode))
+                qstrings.put("code", magicCode);
+            String strUri = String.format("%sapi/usertoken/GetToken", this.uri);
+            URI tokenUrl = null;
+            try {
+                tokenUrl = MakeUri(strUri, qstrings);
+            } catch (URISyntaxException e) {
+                e.printStackTrace();
+                return null;
+            }
+
+            // add botframework api service url to the list of trusted service url's for these app credentials.
+            MicrosoftAppCredentials.trustServiceUrl(tokenUrl.toString());
+
+            // Set Credentials and make call
+            MicrosoftAppCredentials appCredentials = (MicrosoftAppCredentials) client.restClient().credentials();
+
+            // Later: Use client in clientimpl?
+            OkHttpClient client = new OkHttpClient.Builder()
+                    .addInterceptor(new MicrosoftAppCredentialsInterceptor(appCredentials))
+                    .build();
+
+            Request request = new Request.Builder()
+                    .url(tokenUrl.toString())
+                    .header("User-Agent", UserAgent.value())
+                    .build();
+
+            Response response = null;
+            try {
+                response = client.newCall(request).execute();
+                int statusCode = response.code();
+                if (statusCode == HTTP_OK) {
+                    return this.mapper.readValue(response.body().string(), TokenResponse.class);
+                }
+                else if (statusCode == HTTP_NOT_FOUND) {
+                    return null;
+                }
+                else {
+                    return null;
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
```

is eating the exception intended?

---

Yes, failure condition to caller